proto/viam/app/package/v1/package.proto

```diff
@@ -0,0 +1,103 @@
+syntax = "proto3";
+
+package viam.app.package.v1;
+
+import "google/api/annotations.proto";
+import "google/protobuf/struct.proto";
+
+option go_package = "go.viam.com/api/app/package/v1";
+
+service PackageService {
+  // CreatePackage uploads a package to the cloud
+  rpc CreatePackage(stream CreatePackageRequest) returns (CreatePackageResponse) {
+    option (google.api.http) = {
+      post: "/viam/app/package/v1/create"
+    };
+  }
+
+  // DeletePackage removes the given package versions
+  rpc DeletePackage(DeletePackageRequest) returns (DeletePackageResponse) {
+    option (google.api.http) = {
+      delete: "/viam/app/package/v1/delete"
+    };
+  }
+
+  // // GetPackage returns the URL and metadata for a requested package version
+  rpc GetPackage(GetPackageRequest) returns (GetPackageResponse) {
+    option (google.api.http) = {
+      get: "/viam/app/package/v1/package"
+    };
+  }
+
+  // ListPackages get the URLs and metadata for requested packages. Include package name, version, and/or type to filter beyond the required organization_id.
+  rpc ListPackages(ListPackagesRequest) returns (ListPackagesResponse) {
+    option (google.api.http) = {
+      get: "/viam/app/package/v1/list"
+    };
+  }
+}
+
+enum PackageType {
+  PACKAGE_TYPE_UNSPECIFIED = 0;
+  PACKAGE_TYPE_ARCHIVE = 1;
+  PACKAGE_TYPE_ML_MODEL = 2;
+}
+
+message FileInfo {
+  string name = 1;
+  uint64 size = 2;
+}
+
+message PackageInfo {
+  string organization_id = 1;
+  string name = 2;
+  string version = 3;
+  PackageType type = 4;
+  repeated FileInfo files = 5;
+  google.protobuf.Struct metadata = 6;
+}
+
+message CreatePackageRequest {
+  oneof package {
+    PackageInfo info = 1;
+    bytes contents = 2; // .tar.gz file
```

Comment:
[supernit] Can put the comment above the field, since we'll update this comment in a follow-up PR to match the wording in [data_sync.proto](https://github.com/viamrobotics/api/pull/109#discussion_r1047683742)

Reply:
done